File: src/two_sum_1/solution_hash_table.py
from typing import List


class Solution:
    def two_sum(self, nums: List[int], target: int) -> List[int]:
        lookup_table = {}

        for idx, num in enumerate(nums):
            find = target - num

            idx_find = lookup_table.get(find)
            if idx_find is not None:
                return [idx_find, idx]

            lookup_table[num] = idx

        return [-1, -1]


def test():
    print(Solution().two_sum([3, 2, 4], 6))
    print(Solution().two_sum([0, 4, 3, 0], 0))

File: src/two_sum_1/solution_hash_table.rs
use std::collections::HashMap;

pub struct Solution;

impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut lookup_table: HashMap<i32, i32> = HashMap::new();

        for (idx, &num) in nums.iter().enumerate() {
            let idx = idx as i32;
            let find = target - num;

            if let Some(&idx_find) = lookup_table.get(&find) {
                return vec![idx_find, idx];
            }

            lookup_table.insert(num, idx);
        }

        vec![-1, -1]
    }
}

pub fn test() {
    println!("{:?}", Solution::two_sum(vec![3, 2, 4], 6))
}
